src/test/scala/ylabs/messaging/ClientTest.scala

```diff
@@ -149,6 +152,130 @@ class ClientTest extends WordSpec with Matchers with BeforeAndAfterEach {
+
+    user1 ! DeleteUser
+    val connected2 = user1 ? Connect(username, userPass)
+    connected2.value.get.get match {
+      case ConnectError(t) ⇒ //that's all we want to check
+    }
+  }
+
+  "with user domain connects to the xmpp server enables users to chat to each other" in new FixtureWithDomain {
+    withTwoConnectedUsers {
+      user1 ! SendMessage(username2, testMessage)
+      verifyMessageArrived(user2Listener, username1, username2, testMessage)
+    }
+  }
+
+  "with user domain connects to the xmpp server enables async chats (message recipient offline)" in new FixtureWithDomain {
+    withTwoUsers {
+      user1 ! Connect(username1, user1Pass)
+      val user2Listener = newEventListener
+      user2 ! RegisterEventListener(user2Listener.ref)
+
+      user1 ! SendMessage(username2, testMessage)
+
+      // yeah, sleeping is bad, but I dunno how else to make this guaranteed async.
+      Thread.sleep(1000)
+      user2 ! Connect(username2, user2Pass)
+
+      verifyMessageArrived(user2Listener, username1, username2, testMessage)
+    }
+  }
+
+  "with user domain connects to the xmpp server enables XEP-0066 file transfers" in new FixtureWithDomain {
+    withTwoConnectedUsers {
+      val fileUrl = "https://raw.githubusercontent.com/mpollmeier/gremlin-scala/master/README.md"
+      val fileDescription = Some("file description")
+      user1 ! SendFileMessage(username2, fileUrl, fileDescription)
+
+      user2Listener.expectMsgPF(3 seconds, "xep-0066 file transfer") {
+        case FileMessageReceived(chat, message, outOfBandData) ⇒
+          chat.getParticipant should startWith(username1.value)
+          message.getTo should startWith(username2.value)
+          outOfBandData.url shouldBe fileUrl
+          outOfBandData.desc shouldBe fileDescription
+      }
+    }
+  }
+
+  "with user domain connects to the xmpp server informs event listeners about chat partners becoming available / unavailable" in new FixtureWithDomain {
+    withTwoConnectedUsers {
+      user1 ! SendMessage(username2, testMessage)
+      verifyMessageArrived(user2Listener, username1, username2, testMessage)
+      user1Listener.fishForMessage(3 seconds, "notification that user2 came online") {
+        case UserBecameAvailable(user) ⇒
+          user.value should startWith(username2.value)
+          true
+      }
+
+      user2 ! Disconnect
+      user1Listener.fishForMessage(3 seconds, "notification that user2 went offline") {
+        case UserBecameUnavailable(user) ⇒
+          user.value should startWith(username2.value)
+          true
+      }
+
+      user2 ! Connect(username2, user2Pass)
+      user1Listener.fishForMessage(3 seconds, "notification that user2 came online") {
+        case UserBecameAvailable(user) ⇒
+          user.value should startWith(username2.value)
+          true
+      }
+    }
+  }
+
+  "with user domain connects to the xmpp server provides information about who is online and offline (roster)" in new FixtureWithDomain {
```

Comment:
if all your tests start with the same words it's a sign that you should have a hierarchy of tests. e.g.
```
  "bla" when {
    "foo" should {
      "blub" can {
        "bar" in new Fixture {
```
http://doc.scalatest.org/2.0/index.html#org.scalatest.WordSpec